libgraph/src/analytics/k_shortest_paths/ksssp.cpp

```diff
@@ -0,0 +1,462 @@
+            push_wrap(ctx, dst, 1);
+          }
+        }
+      },
+      katana::wl<WL>(), katana::loopname("runBFS"),
+      katana::disable_conflict_detection());
+
+  if (graph->template GetData<NodeCount>(report_node) == 0) {
+    return false;
+  }
+
+  katana::do_all(katana::iterate(*graph), [&graph](GNode n) {
+    graph->template GetData<NodeCount>(n) = 0;
+  });
+
+  return true;
+}
+
+template <typename GraphTy>
+bool
+CheckReachabilitySync(
+    GraphTy* graph, const typename GraphTy::Node& source,
+    uint32_t report_node) {
+  using GNode = typename GraphTy::Node;
+
+  katana::InsertBag<GNode> current_bag;
+  katana::InsertBag<GNode> next_bag;
+
+  current_bag.push(source);
+  graph->template GetData<NodeCount>(source) = 1;
+
+  while (current_bag.begin() != current_bag.end()) {
+    katana::do_all(
+        katana::iterate(current_bag),
+        [&](GNode n) {
+          for (auto edge : Edges(*graph, n)) {
+            auto dest = EdgeDst(*graph, edge);
+            if (graph->template GetData<NodeCount>(dest) == 0) {
+              graph->template GetData<NodeCount>(dest) = 1;
+              next_bag.push(dest);
+            }
+          }
+        },
+        katana::steal());
+
+    current_bag.clear();
+    std::swap(current_bag, next_bag);
+  }
+
+  if (graph->template GetData<NodeCount>(report_node) == 0) {
+    return false;
+  }
+
+  katana::do_all(katana::iterate(*graph), [&graph](GNode n) {
+    graph->template GetData<NodeCount>(n) = 0;
+  });
+
+  return true;
+}
+
+//delta stepping implementation for finding a shortest path from source to report node
+template <
+    typename GraphTy, typename Item, typename OBIMTy, typename PushWrap,
+    typename EdgeRange>
+void
+DeltaStepAlgo(
+    GraphTy* graph, const typename GraphTy::Node& source,
+    const PushWrap& push_wrap, const EdgeRange& edge_range,
+    katana::InsertBag<std::pair<Weight, Path*>>* report_paths_bag,
+    katana::InsertBag<Path*>* path_pointers, PathAlloc& path_alloc,
+    uint32_t report_node, uint32_t num_paths, uint32_t step_shift) {
+  using GNode = typename GraphTy::Node;
+
+  using kSSSP = katana::analytics::KSsspImplementationBase<
+      GraphTy, Weight, const Path, true>;
+  using kSSSPUpdateRequestIndexer = typename kSSSP::UpdateRequestIndexer;
+
+  using OBIM =
+      katana::OrderedByIntegerMetric<kSSSPUpdateRequestIndexer, PSchunk>;
+
+  //! [reducible for self-defined stats]
+  katana::GAccumulator<size_t> bad_work;
+  //! [reducible for self-defined stats]
+  katana::GAccumulator<size_t> wl_empty_work;
+
+  graph->template GetData<NodeCount>(source) = 1;
+
+  katana::InsertBag<Item> init_bag;
+
+  Path* path = path_alloc.NewPath();
+  path->last = nullptr;
+  path->parent = source;
+
+  path_pointers->push(path);
+
+  push_wrap(init_bag, source, 0, path, "parallel");
+
+  katana::for_each(
+      katana::iterate(init_bag),
+      [&](const Item& item, auto& ctx) {
+        for (auto ii : edge_range(item)) {
```

Comment:
same here: replace by `Edges` function

Reply:
Same as above, but code is [here](https://github.com/dqawami/katana/blob/1d88284a0f19755e80553ad453a2971142e1fae2/libgraph/src/analytics/k_shortest_paths/ksssp.cpp#L351)